Formula/crystal.rb

```diff
@@ -45,14 +45,14 @@ class Crystal < Formula
 
   resource "boot" do
     on_macos do
-      url "https://github.com/crystal-lang/crystal/releases/download/0.36.1/crystal-0.36.1-1-darwin-x86_64.tar.gz"
-      version "0.36.1-1"
-      sha256 "3b381d4ac1241514d63b1f243ed183a1b10945b0e77c5dd625067e91a657ef75"
+      url "https://github.com/crystal-lang/crystal/releases/download/1.0.0/crystal-1.0.0-1-darwin-x86_64.tar.gz"
+      version "1.0.0-1"
+      sha256 "29019828d32df9807a3f08fb7354fa0e44a5ad8b000eccd1dea114f891cbc006"
     end
     on_linux do
-      url "https://github.com/crystal-lang/crystal/releases/download/0.36.1/crystal-0.36.1-1-linux-x86_64.tar.gz"
-      version "0.36.1-1"
-      sha256 "38cc7514f8d9e463665ebaf7232d3a6c46a7397fc9ff5c05fd0f9e4706febb18"
+      url "https://github.com/crystal-lang/crystal/releases/download/1.0.0/crystal-1.0.0-1-linux-x86_64.tar.gz"
+      version "1.0.0-1"
+      sha256 "00211ca77758e99210ec40b8c5517b086d2ff9909e089400f6d847a95e5689a4"
     end
   end
```

Comment:
Any reason why we're bootstrapping with the last version of `crystal` here? I realise that's what we did on the last version bump, but I don't know if that was just an oversight or if it was intentional.

Reply:
I'm not sure there's a particular reason, except that it was always like this 🤷 Any insights @bcardiff?

In this particular case, it certainly makes some sense to boot from Crystal 1.0 as it's the first stable release. However, there is no guarantee that any future release can be built from any earlier release.

Reply:
The change is intended. Every new crystal release is built from the previous one. Although in some cases some version can be skip, the rule is that a compiler is built with the previous minor version.

Reply:
>  I realise that's what we did on the last version bump

It's what we always did, not just for the last version bump. The exceptions are when we make a minor release (only bug fixes), mainly because those bugs could make the compiler from stopping compiling the next compiler.